tests/system/test_aws_s3.py

```diff
@@ -0,0 +1,104 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import random
+
+import pandas as pd
+import pytest
+
+import mlrun
+import mlrun.errors
+import mlrun.feature_store as fstore
+from mlrun.datastore.datastore_profile import (
+    DatastoreProfileS3,
+    register_temporary_client_datastore_profile,
+)
+from mlrun.datastore.targets import ParquetTarget
+from mlrun.utils import logger
+from tests.system.base import TestMLRunSystem
+
+# Used to test dataframe functionality (will be saved as csv)
+test_df_string = "col1,col2,col3\n1,2,3"
+
+test_environment = TestMLRunSystem._get_env_from_file()
+aws_keys = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_BUCKET_NAME"]
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.skipif(
+    not all(test_environment[key] for key in aws_keys),
+    reason=f"one of {aws_keys} keys is not set, skipping until testing against real aws",
+)
+@pytest.mark.parametrize("use_datastore_profile", [True, False])
+class TestAwsS3(TestMLRunSystem):
+    def custom_setup(self):
+        pass
+
+    def _make_target_names(
+        self, prefix, bucket_name, object_dir, object_file, csv_file
+    ):
+        bucket_path = prefix + bucket_name
+        object_path = f"{object_dir}/{object_file}"
+        df_path = f"{object_dir}/{csv_file}"
+        object_url = f"{bucket_path}/{object_path}"
+        res = {
+            "bucket_path": bucket_path,
+            "object_path": object_path,
+            "df_path": df_path,
+            "object_url": object_url,
+            "df_url": f"{bucket_path}/{df_path}",
+            "blob_url": f"{object_url}.blob",
+            "parquet_url": f"{object_url}.parquet",
+        }
+        return res
+
+    def setup_method(self, method):
+        super().setup_method(method)
+        self._bucket_name = test_environment["AWS_BUCKET_NAME"]
+        self._access_key_id = test_environment["AWS_ACCESS_KEY_ID"]
+        self._secret_access_key = test_environment["AWS_SECRET_ACCESS_KEY"]
+
+        object_dir = "test_mlrun_s3_objects"
+        object_file = f"file_{random.randint(0, 1000)}.txt"
+        csv_file = f"file_{random.randint(0,1000)}.csv"
+
+        self.s3 = {}
+        self.s3["s3"] = self._make_target_names(
+            "s3://", self._bucket_name, object_dir, object_file, csv_file
+        )
+        self.s3["ds"] = self._make_target_names(
+            "ds://s3ds_profile/", self._bucket_name, object_dir, object_file, csv_file
+        )
```

Comment:
<img width="735" alt="Screenshot 2023-09-18 at 13 49 38" src="https://github.com/mlrun/mlrun/assets/4675524/f85af896-3568-4a93-918a-c7d114b78364">
Not a linter-enforced style rule, but I think they have a point.

Reply:
will do

Reply:
will do